wasm_filter/src/lib.rs

```diff
@@ -4,65 +4,97 @@ use proxy_wasm::traits::*;
 use proxy_wasm::types::*;
 use std::time::Duration;
 
+mod config;
+
+const AUTH_BACKEND: &str = "httpbin";
+
 #[no_mangle]
 pub fn _start() {
     proxy_wasm::set_log_level(LogLevel::Trace);
     proxy_wasm::set_http_context(|context_id, _| -> Box<dyn HttpContext> {
         Box::new(HttpHeaders { context_id })
     });
-    proxy_wasm::set_root_context(|_| -> Box<dyn RootContext> { Box::new(HelloWorld) });
+    proxy_wasm::set_root_context(|_| -> Box<dyn RootContext> { Box::new(ConfigContext) });
 }
 
 struct HttpHeaders {
     context_id: u32,
 }
 
-impl Context for HttpHeaders {}
+struct ConfigContext;
 
-impl HttpContext for HttpHeaders {
-    fn on_http_request_headers(&mut self, _: usize) -> Action {
-        for (name, value) in &self.get_http_request_headers() {
-            info!("#{} -> {}: {}", self.context_id, name, value);
-        }
+impl Context for ConfigContext {}
 
-        match self.get_http_request_header(":path") {
-            Some(path) if path == "/hello" => {
-                self.send_http_response(
-                    200,
-                    vec![("Hello", "World"), ("Powered-By", "proxy-wasm")],
-                    Some(b"Hello, World!\n"),
-                );
-                Action::Pause
-            }
-            _ => Action::Continue,
-        }
+impl RootContext for ConfigContext {
+    fn on_vm_start(&mut self, _: usize) -> bool {
+        let config = self.get_configuration();
+        config::import_config(std::str::from_utf8(&config.unwrap()).unwrap());
+        self.set_tick_period(Duration::from_secs(20));
+        true
     }
 
-    fn on_http_response_headers(&mut self, _: usize) -> Action {
-        for (name, value) in &self.get_http_response_headers() {
-            info!("#{} <- {}: {}", self.context_id, name, value);
-        }
-        Action::Continue
+    fn on_tick(&mut self) {
+        let datetime: DateTime<Utc> = self.get_current_time().into();
+        log::debug!("Wasm filter tick: {}", datetime);
     }
+}
 
-    fn on_log(&mut self) {
-        info!("#{} completed.", self.context_id);
+impl Context for HttpHeaders {
+    fn on_http_call_response(&mut self, _: u32, _: usize, _: usize, _: usize) {
+        let headers = self.get_http_call_response_headers();
+        for (name, value) in &headers {
+            if name == &":status".to_string() && value == &"200".to_string() {
```

Comment:
This is the wrong way to do this comparison, because you are doing a lot of extra work: allocating and copying in a couple of strings, comparing them, and then deallocating them.

Each `String` owns its contents as a string slice `&str`, so instead of comparing `String`s you can compare slices. Obtaining a string slice from a `String` is direct and free via `as_str()` (does not do anything expensive, like allocating) and once you have that, you can compare them directly with your own static string slices, such as `":status"` and `"200"`. When you are done with the comparisons, your string slices can be dropped, but because they are borrowed (references are `Copy`) there is no cleanup to perform, so no deallocation of a temporary allocation happens.

`if name.as_str() == ":status" && value.as_str() == "200" {`